code/game/machinery/poolcontroller.dm

```diff
@@ -39,63 +42,73 @@
 	updatePool() //Call the mob affecting/decal cleaning proc
 
 /obj/machinery/poolcontroller/proc/updatePool()
-	for(var/turf/simulated/floor/beach/water/W in linkedturfs) //Check for pool-turfs linked to the controller.
-		for(var/mob/M in W) //Check for mobs in the linked pool-turfs.
-			//Sanity checks, don't affect robuts, AI eyes, and observers
-			if(isAIEye(M))
-				continue
-			if(issilicon(M))
-				continue
-			if(isobserver(M))
-				continue
-			if(M.stat == DEAD)
-				continue
-			//End sanity checks, go on
-			switch(temperature) //Apply different effects based on what the temperature is set to.
-				if("scalding") //Burn the mob.
-					M.bodytemperature = min(500, M.bodytemperature + 35) //heat mob at 35k(elvin) per cycle
-					M << "<span class='danger'>The water is searing hot!</span>"
-
-				if("warm") //Gently warm the mob.
-					M.bodytemperature = min(330, M.bodytemperature + 10) //Heats up mobs to just over normal, not enough to burn
-					if(prob(50)) //inform the mob of warm water half the time
-						M << "<span class='warning'>The water is quite warm.</span>" //Inform the mob it's warm water.
-
-				if("cool") //Gently cool the mob.
-					M.bodytemperature = max(290, M.bodytemperature - 10) //Cools mobs to just below normal, not enough to burn
-					if(prob(50)) //inform the mob of cold water half the time
-						M << "<span class='warning'>The water is chilly.</span>" //Inform the mob it's chilly water.
-
-				if("frigid") //Freeze the mob.
-					M.bodytemperature = max(80, M.bodytemperature - 35) //cool mob at -35k per cycle
-					M << "<span class='danger'>The water is freezing!</span>"
-
+	for(var/turf/T in linkedturfs) //Check for pool-turfs linked to the controller.
+		for(var/mob/M in T) //Check for mobs in the linked pool-turfs.
+			if(temperature != "normal")		//don't bother with this if we aren't going to be changing anything anyways
```

Comment:
No clue, why it's not in there, but "normal" does nothing so I figured there was no point in even calling the proc if we won't be using it.